streaming/base/dataset.py

```diff
@@ -313,6 +327,22 @@ def __init__(self,
             }
             for stream in streams:
                 stream.apply_default(default)
+        elif epoch_size_value:
```

Comment:
I think you might want to revert this to the older PR approach where you included this logic in below `else` section !!!

Reply:
yep my mistake. bad merge